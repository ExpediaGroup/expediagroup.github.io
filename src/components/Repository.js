/*
Copyright 2022 Expedia, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from "react";
import styles from "./Repository.module.css";
import clsx from "clsx";
import ArrowLink from "./ArrowLink";

function Repository({name, description, imageUrl, repoUrl}) {
    return (
        <div className={clsx('col col--4', styles.repository)}>
            <div className="text--center">
                <a href={repoUrl} target="_blank">
                    <img className={styles.repositoryImage} src={imageUrl} alt={name} />
                </a>
            </div>
            <div className={styles.repositoryTitle}>
                <a href={repoUrl} target="_blank">
                    <h3>{name}</h3>
                </a>
                <ArrowLink link={repoUrl}/>
            </div>
            <p className={styles.repositoryDescription}>{description}</p>
        </div>
    );
}

export default Repository;
