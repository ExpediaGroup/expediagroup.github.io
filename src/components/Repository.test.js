/*
Copyright 2022 Expedia, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import renderer from 'react-test-renderer';
import Repository from './Repository';

jest.mock('./ArrowLink', () => 'ArrowLink');

const NAME = 'my-repo';
const DESCRIPTION = 'my description';
const IMAGE_URL = 'https://my/repo/image';
const REPO_URL = 'https://my/repo';

it('renders correctly', () => {
    const tree = renderer
        .create(<Repository name={NAME} description={DESCRIPTION} imageUrl={IMAGE_URL} repoUrl={REPO_URL}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
