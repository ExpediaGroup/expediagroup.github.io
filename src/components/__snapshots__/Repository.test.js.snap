// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col col--4 repository"
>
  <div
    className="text--center"
  >
    <a
      href="https://my/repo"
      target="_blank"
    >
      <img
        alt="my-repo"
        className="repositoryImage"
        src="https://my/repo/image"
      />
    </a>
  </div>
  <div
    className="repositoryTitle"
  >
    <a
      href="https://my/repo"
      target="_blank"
    >
      <h3>
        my-repo
      </h3>
    </a>
    <ArrowLink
      link="https://my/repo"
    />
  </div>
  <p
    className="repositoryDescription"
  >
    my description
  </p>
</div>
`;
